boilerplate/openshift/golang-osd-operator/standard.mk

```diff
@@ -191,15 +209,30 @@ openapi-generate:
 .PHONY: generate
 generate: op-generate go-generate openapi-generate
 
+
 .PHONY: go-build
 go-build: ## Build binary
 	# Force GOOS=linux as we may want to build containers in other *nix-like systems (ie darwin).
 	# This is temporary until a better container build method is developed
 	${GOENV} GOOS=linux go build ${GOBUILDFLAGS} -o ${BINFILE} ${MAINPACKAGE}
 
+# ENVTEST_K8S_VERSION refers to the version of kubebuilder assets to be downloaded by envtest binary.
+ENVTEST_K8S_VERSION = 1.23
+
+ENVTEST = $(shell pwd)/bin/setup-envtest
+.PHONY: envtest
+envtest: ## Download envtest-setup locally if necessary.
+	$(call go-get-tool,$(ENVTEST),sigs.k8s.io/controller-runtime/tools/setup-envtest@latest)
+	
+# Setting SHELL to bash allows bash commands to be executed by recipes.
+# This is a requirement for 'setup-envtest.sh' in the test target.
+# Options are set to exit when a recipe line exits non-zero or a piped command fails.
+SHELL = /usr/bin/env bash -o pipefail
+.SHELLFLAGS = -ec
+
 .PHONY: go-test
-go-test:
-	${GOENV} go test $(TESTOPTS) $(TESTTARGETS)
+go-test: envtest
+	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)" go test ./... -coverprofile cover.out
```

Comment:
I don't think this should be using `-coverprofile`. Leave that for the [`coverage` target](https://github.com/openshift/boilerplate/pull/211/files#diff-aef0082c5bb612084a57562263d822bb2a2a089ec35cfcf3feec179673c56b11R280), which operates via [codecov.sh](https://github.com/openshift/boilerplate/blob/bea9ec5ce249a5a7182598bbb4fea5ca148bbc5f/boilerplate/openshift/golang-osd-operator/codecov.sh) invoking `make go-test` with certain TESTOPTS.

Reply:
thanks, this was copy&past from the Makefile generated by latest osdk.